services/drupal/web/modules/custom/epa_media/epa_media.install

```diff
@@ -5,3 +5,131 @@
 function epa_media_update_9000() {
   \Drupal::database()->query("UPDATE {file_managed} SET filemime = 'application/json' WHERE filename LIKE '%.json' AND filemime = 'application/octet-stream';");
 }
+
+/**
+ * Move all json and xml "Document" media entities to be "Other" media bundle
+ */
+function epa_media_update_9001(&$sandbox) {
+  $database = \Drupal::database();
+
+  if (!isset($sandbox['total'])) {
+    // Find all "Document" media entities that have a .json or .xml file.
+    $media_ids = $database
+      ->query("SELECT entity_id
+        FROM {media__field_media_file}
+        WHERE field_media_file_target_id IN (
+	          SELECT fid
+	          FROM file_managed
+	          WHERE filename like '%.json' OR filename like '%.xml'
+        )
+        AND
+        bundle = 'document'")
+      ->fetchCol('entity_id');
+
+    $sandbox['total'] = count($media_ids);
+    $sandbox['current'] = 0;
+  }
+
+  // Do not continue if no media items are found.
+  if (empty($sandbox['total'])) {
+    $sandbox['#finished'] = 1;
+    return t('No Media entities to be processed.');
+  }
+
+  // Arbitrary limit to process 50 at a time.
+  $ids = array_slice($media_ids, $sandbox['current'], 50);
+
+  foreach ($ids as $id) {
```

Comment:
I think we can avoid actually iterating over each record and each of these queries could be rewritten with IN criteria that matches on a subquery.   See an example below.

Reply:
I've now updated all queries to use arrays with IN and works as expected.